### PR TITLE
Convert leaflet declaration to UMD export

### PR DESCRIPTION
**Pull request by Jessidhia**
Use the `export as namespace` syntax to avoid having to use triple-slash imports and to be able to do more natural exports.

This might need some refactoring.

Should help with https://github.com/DefinitelyTyped/DefinitelyTyped/issues/17979

### CODE REVIEW

File: types/leaflet/index.d.ts [hunk 1]
```diff
@@ -3,1549 +3,1546 @@
...
+    Polygon as GeoJSONPolygon,
+    MultiPolygon as GeoJSONMultiPolygon,
+    FeatureCollection as GeoJSONFeatureCollection,
+    GeometryCollection as GeoJSONGeometryCollection,
+    Point as GeoJSONPoint,
+    MultiPoint as GeoJSONMultiPoint,
+    GeoJsonObject as GeoJSONGeoJsonObject
+} from 'geojson'
+
+export as namespace L
```
Comment (Kovensky) [inline types/leaflet/index.d.ts]: Annoyingly, it doesn't do that through an UMD wrapper, but it just unconditionally adds its own (ES module) `exports` object to `window.L`. I'm not sure that even compiles with webpack >= 2, and definitely wouldn't work with real modules.

**Comment by Kovensky** [comment]
Looks like this will need a bigger refactoring than planned. It will take me a while to update this such that tests pass.

**Comment by alejo90** [comment]
First of all, thanks for your contribution. If you can contribute with a large-ish refactor, it would be better to convert the definition to es2015 modules, since leaflet 1.1.0 is out now and that's what they did. The definition was written the way it was beacuse it was the right way to represent the version of leaflet it targeted as it was written and with the version of Typescript available at the time. The other definition authors and I knew what we were doing. The landscape has evolved since then and updates are required but converting this to an UMD export is certainly not one of them.

**Comment by Kovensky** [comment]
Yeah, I’m a relative newcomer so I’ve only been in a world with strict null checks and UMD declarations; after realizing it was probably older than 2.0 I understood why it was written the way it was.

I’m not sure what to do about the UMD export though; leaflet on master still puts its own exports object on window.L.

**Comment by RyanCavanaugh** [comment]
@Kovensky Please fix the failures indicated in the Travis CI log.

**Comment by mhegazy** [comment]
@Kovensky looks like this one has gone stale. closing "abandon" PRs as usual. reopen if you still wanna pursue this change.

**Comment by aluanhaddad** [comment]
@Kovensky do you plan to continue resume work on this? It would be a very good change.

**Pull request closed without merging**